docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -14,19 +14,63 @@ helpviewer_keywords:
+| Allow non-string JSON values for string properties  | [Not supported](#non-string-values-for-string-properties) |
+| Custom converter registration                       | [Order of precedence differs](#converter-registration-precedence) |
+| Minimal character escaping                          | [Strict character escaping, configurable](#character-escaping) |
+| Deserialize inferred type to `object` properties    | [Requires a custom converter, sample provided](#deserialization-of-object-properties) |
+| No maximum depth by default                         | [Default maximum depth 64, configurable](#maximum-depth) |
+| `DateTimeZoneHandling`, `DateFormatString` settings | [Requires a custom converter, sample provided](#specify-date-format) |
+| Deserialize strings as numbers                      | [Requires a custom converter, sample provided](#quoted-numbers) |
+| Deserialize `Dictionary` with non-string key        | [Requires a custom converter, sample provided](#dictionary-with-non-string-key) |
+| Polymorphic serialization                           | [Requires a custom converter, sample provided](#polymorphic-serialization) |
+| Polymorphic deserialization                         | [Requires a custom converter, sample provided](#polymorphic-serialization) |
+| Required properties                                 | [Requires a custom converter, sample provided](#required-properties) |
+| Deserialize JSON `null` literal to non-nullable types | [Requires a custom converter, sample provided](#deserialize-null-to-non-nullable-type) |
+| Deserialize to immutable classes and structs        | [Requires a custom converter, sample provided](#deserialize-to-immutable-classes-and-structs) |
+| `[JsonConstructor]` attribute                       | [Requires a custom converter, sample provided](#specify-constructor-to-use) |
+| Use `DefaultContractResolver` to include or exclude properties | [Requires a custom converter, sample provided](#conditionally-ignore-a-property) |
+| `NullValueHandling.Ignore` global setting           | [IgnoreNullValues global option](#omit-null-value-properties) |
+| `NullValueHandling.Ignore` setting on `[JsonProperty]` attribute | [Requires a custom converter, sample provided](#conditionally-ignore-a-property)  |
+| `DefaultValueHandling` global setting               |[Requires a custom converter, sample provided](#conditionally-ignore-a-property) |
+| `DefaultValueHandling` setting on `[JsonProperty]` attribute | [Requires a custom converter, sample provided](#conditionally-ignore-a-property)  |
+| Callbacks                                           | [Requires a custom converter, sample provided](#callbacks) |
+| Support for a broad range of types                  | [Some types require custom converters](#types-without-built-in-support) |
+| Support for public and non-public fields            | [Requires a custom converter](#public-and-non-public-fields) |
```

Comment:
Other type of ordering we could consider is what's more common. In that case, items like these should be more on the top where as some of the other/less used features at the bottom. What were you thinking in terms of sort order?

Reply:
Also, I thought things like these are in the "not supported" column (they are in the `Scenarios that JsonSerializer currently doesn't support` heading), no?

Reply:
The overall order now is by groups: supported, workaround with sample, workaround without sample, no workaround practical. Within each grouping ideally we'd go from more common to less common, but it's not always easy to differentiate on that basis.  I'm open to suggestions on changing the order.

The Scenarios that JsonSerializer currently doesn't support section originally was two sections that got conflated -- workaround-no-sample and workaround-not-practical. We could split it up again, which would more closely match the groupings in the table.

Reply:
> The overall order now is by groups: supported, workaround with sample, workaround without sample, no workaround practical. Within each grouping ideally we'd go from more common to less common, but it's not always easy to differentiate on that basis.

Perfect.

> I'm open to suggestions on changing the order.

Provided some thoughts. Thanks.

> We could split it up again, which would more closely match the groupings in the table.

This doc will likely keep evolving, especially until we ship 5.0. So, let's leave that change out for now.